server/controllers/chartsController/helpers.test.js

```diff
@@ -383,6 +383,152 @@ describe('chartsController - helpers', () => {
         },
       })
     })
+
+    it('it replaces site abbreviation for the site name', () => {
+      const studyTotals = {
+        Madrid: { count: 10, targetTotal: undefined },
+        Melbourne: { count: 35, targetTotal: undefined },
+        NORTHWELL: { count: 5, targetTotal: undefined },
+        UNC: { count: 0, targetTotal: undefined },
+        [helpers.TOTALS_STUDY]: { count: 50, targetTotal: undefined },
+      }
+      const data = new Map([
+        ['Madrid-male-undefined', 5],
+        ['Madrid-female-undefined', 5],
+        ['Melbourne-male-undefined', 10],
+        ['Melbourne-female-undefined', 25],
+        ['NORTHWELL-male-undefined', 5],
+        ['NORTHWELL-female-undefined', 0],
+        ['UNC-male-undefined', 0],
+        ['UNC-female-undefined', 0],
+        [`${helpers.TOTALS_STUDY}-male`, 20],
+        [`${helpers.TOTALS_STUDY}-female`, 30],
+      ])
```

Comment:
It's awesome to see tests here 🙌 
looks like we define this `data` object a few times in this file. If they are always the same should it be defined once at the top? If they need to be slightly different in each case you could use a factory here so you only need to specify the parts relevant to each test. Here's [an example](https://github.com/TheGnarCo/gridly/blob/main/app/javascript/src/testUtils/factories.ts#L238-L247) of that pattern.

Reply:
here's [an example usage](https://github.com/TheGnarCo/gridly/blob/main/app/javascript/src/pages/projects/HVACPage/HVACPage.spec.tsx#L29)

Reply:
Here is an adaptation of that for a Map:
```js
const factory = (overrides = {}) => {
    const mapEntries = {
        foo: 'foo',
        bar: 'bar',
        ...overrides
    }
    
    return new Map(Object.entries(mapEntries))
}

/*
: factory()
> Map(2) {'foo' => 'foo', 'bar' => 'bar'}

: factory({bar: 'hello world'})
> Map(2) {'foo' => 'foo', 'bar' => 'hello world'}
*/
```